### PR TITLE
1 developing landing

### LINKED ISSUE #1: Developing Landing

### PR DESCRIPTION
**Pull request by behkha**

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -4,10 +4,12 @@
     <meta charset="UTF-8">
     <link rel="icon" href="/favicon.ico">
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
-    <title>Vite App</title>
+    <title>Jeodns</title>
+    <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.3.0-alpha1/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-GLhlTQ8iRABdZLl6O3oVMWSktQOp6b7In1Zl3/Jr59b6EGGoI1aFkw7cmDA6j6gD" crossorigin="anonymous">
```
Comment (yeganemehr) [inline index.html]: remove external files.
if you need bootstrap use npm

File: src/assets/main.css [hunk 2]
```diff
@@ -1,35 +1,109 @@
 @import './base.css';
 
-#app {
-  max-width: 1280px;
-  margin: 0 auto;
-  padding: 2rem;
+@font-face {
```
Comment (yeganemehr) [inline src/assets/main.css:3]: just use one font, either IRANSansWeb or IRANSansWeb(FaNum)

File: src/assets/main.css [hunk 3]
```diff
@@ -1,35 +1,109 @@
...
+  font-family: "Black";
```
Comment (yeganemehr) [inline src/assets/main.css]: use different variant of front with same font-family but with different weight and style

File: src/components/Brand.vue [hunk 4]
```diff
@@ -0,0 +1,65 @@
+<template>
```
Comment (yeganemehr) [inline src/components/Brand.vue:1]: Make it a link to https://jeodns.com

File: index.html [hunk 5]
```diff
@@ -4,10 +4,12 @@
     <meta charset="UTF-8">
     <link rel="icon" href="/favicon.ico">
```
Comment (yeganemehr) [inline index.html:5]: use proper favicon based `Brand` component.

File: src/components/DrawerMenu.vue [hunk 6]
```diff
@@ -0,0 +1,79 @@
+<template>
+  <div class="drawer-menu__container mx-4 pt-3">
+    <div class="d-flex flex-row justify-content-between">
+      <img src="@/assets/icons/menu-burger.svg" alt="" data-bs-toggle="offcanvas" data-bs-target="#offcanvasRight" aria-controls="offcanvasRight">
+      <Brand />
+    </div>
+    <div class="drawer-menu__offcanvas offcanvas offcanvas-end" tabindex="-1" id="offcanvasRight" aria-labelledby="offcanvasRightLabel">
+      <div class="drawer-menu__items d-flex flex-column align-items-end mx-4 mt-4">
+        <MenuItem title="هلدینگ داده نگار جی" color="#173D7A">
```
Comment (yeganemehr) [inline src/components/DrawerMenu.vue]: link it to https://dnj.co.ir

File: src/components/DrawerMenu.vue [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+          <template #icon>
+            <img src="@/assets/icons/left-chevron-dark.svg" alt="" />
+          </template>
+        </MenuItem>
+        <MenuItem title="جی سرور" color="#173D7A">
```
Comment (yeganemehr) [inline src/components/DrawerMenu.vue]: link it to https://jeyserver.com

File: src/components/DrawerMenu.vue [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+          <template #icon>
+            <img src="@/assets/icons/left-chevron-dark.svg" alt="" />
+          </template>
+        </MenuItem>
+        <MenuItem title="وب شات" color="#173D7A">
```
Comment (yeganemehr) [inline src/components/DrawerMenu.vue]: link it to https://web-shot.ir

File: src/components/DrawerMenu.vue [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+          <template #icon>
+            <img src="@/assets/icons/left-chevron-dark.svg" alt="" />
+          </template>
+        </MenuItem>
+        <MenuItem title="JeoIP" color="#173D7A">
```
Comment (yeganemehr) [inline src/components/DrawerMenu.vue]: link it to https://jeoip.ir

File: src/components/DrawerMenu.vue [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+          <template #icon>
+            <img src="@/assets/icons/left-chevron-dark.svg" alt="" />
+          </template>
+        </MenuItem>
+        <div class="drawer-menu__divider"></div>
+        <MenuItem title="پشتیبانی" color="#173D7A">
```
Comment (yeganemehr) [inline src/components/DrawerMenu.vue]: link it to https://jeyserver.com/fa/userpanel/ticketing

File: src/components/Footer.vue [hunk 11]
```diff
@@ -0,0 +1,48 @@
+<template>
+  <div class="footer__container d-flex flex-row justify-content-end mx-5 mt-3">
+    <div class="footer-items__container d-flex flex-row my-auto mx-3">
+      <div class="footer__item d-flex flex-column align-items-center justify-items-center">
```
Comment (yeganemehr) [inline src/components/Footer.vue]: Make a tooltip for it
also link it to https://t.me/dnjco

File: src/components/Footer.vue [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+        <img class="m-auto" src="@/assets/icons/telegram.svg" alt="">
+      </div>
+      <div class="footer__item d-flex flex-column align-items-center justify-items-center">
```
Comment (yeganemehr) [inline src/components/Footer.vue]: make a tooltip for it.
also link it to mailto:hi@dnj.co.ir

File: src/components/Footer.vue [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+        <img class="m-auto" src="@/assets/icons/mail.svg" alt="">
+      </div>
+      <div class="footer__item d-flex flex-column align-items-center justify-items-center">
```
Comment (yeganemehr) [inline src/components/Footer.vue]: make a tooltip for it
also link it to tel:03134420301

File: src/components/ContactCard.vue [hunk 14]
```diff
@@ -0,0 +1,99 @@
+<template>
+  <div class="contact-card__container px-4">
+    <p class="contact-card__title bold-font mb-4">ما برای همکاری با شما آماده ایم؛ <br/>  با ما در تماس باشید!</p>
+    <input type="text" class="contact-card__input regular-font form-control" placeholder="نام شما">
+    <input type="number" class="contact-card__input mt-4 regular-font form-control" placeholder="شماره تماس">
```
Comment (yeganemehr) [inline src/components/ContactCard.vue]: use a text input

File: src/components/ContactCard.vue [hunk 15]
```diff
@@ -0,0 +1,99 @@
+<template>
+  <div class="contact-card__container px-4">
```
Comment (yeganemehr) [inline src/components/ContactCard.vue]: make it a form with validation and actual ajax request.
name must have more 5 characters.
phone number must be valid 11 digit phone number (0{prefix: 2}{phone: 8})
message must have 10 characters or more.

File: src/components/ContactCard.vue [hunk 16]
```diff
@@ -0,0 +1,99 @@
...
+    <p class="contact-card__title bold-font mb-4">ما برای همکاری با شما آماده ایم؛ <br/>  با ما در تماس باشید!</p>
+    <input type="text" class="contact-card__input regular-font form-control" placeholder="نام شما">
+    <input type="number" class="contact-card__input mt-4 regular-font form-control" placeholder="شماره تماس">
+    <textarea class="contact-card__textarea form-control mt-4 regular-font" placeholder="پیام شما" rows="6"></textarea>
+    <button class="contact-card__submit py-2 mt-4 d-flex flex-row justify-content-center" data-bs-toggle="modal" data-bs-target="#staticBackdrop">
+      <p class="medium-font mx-1 my-auto">ارسال پیام</p>
+      <img class="my-auto" src="@/assets/icons/arrow-left.svg" alt="">
```
Comment (yeganemehr) [inline src/components/ContactCard.vue]: use material design icons.

File: src/components/DrawerMenu.vue [hunk 17]
```diff
@@ -0,0 +1,79 @@
...
+  <div class="drawer-menu__container mx-4 pt-3">
+    <div class="d-flex flex-row justify-content-between">
+      <img src="@/assets/icons/menu-burger.svg" alt="" data-bs-toggle="offcanvas" data-bs-target="#offcanvasRight" aria-controls="offcanvasRight">
+      <Brand />
+    </div>
+    <div class="drawer-menu__offcanvas offcanvas offcanvas-end" tabindex="-1" id="offcanvasRight" aria-labelledby="offcanvasRightLabel">
+      <div class="drawer-menu__items d-flex flex-column align-items-end mx-4 mt-4">
+        <MenuItem title="هلدینگ داده نگار جی" color="#173D7A">
+          <template #icon>
+            <img src="@/assets/icons/left-chevron-dark.svg" alt="" />
```
Comment (yeganemehr) [inline src/components/DrawerMenu.vue]: use material design icons.

File: src/components/DrawerMenu.vue [hunk 18]
```diff
@@ -0,0 +1,79 @@
+<template>
+  <div class="drawer-menu__container mx-4 pt-3">
+    <div class="d-flex flex-row justify-content-between">
+      <img src="@/assets/icons/menu-burger.svg" alt="" data-bs-toggle="offcanvas" data-bs-target="#offcanvasRight" aria-controls="offcanvasRight">
```
Comment (yeganemehr) [inline src/components/DrawerMenu.vue]: use material design icons.

File: src/components/Menu.vue [hunk 19]
```diff
@@ -0,0 +1,44 @@
+<template>
+  <div class="d-flex flex-row justify-space-around">
+    <MenuItem title="پشتیبانی">
+      <template #icon>
+        <img src="@/assets/icons/headset_mic.svg" alt="" />
```
Comment (yeganemehr) [inline src/components/Menu.vue]: use material design icons.

**Review by yeganemehr — CHANGES_REQUESTED** [review]

File: src/components/ContactCard.vue [hunk 20]
```diff
@@ -0,0 +1,99 @@
+<template>
+  <div class="contact-card__container px-4">
+    <p class="contact-card__title bold-font mb-4">ما برای همکاری با شما آماده ایم؛ <br/>  با ما در تماس باشید!</p>
+    <input type="text" class="contact-card__input regular-font form-control" placeholder="نام شما">
+    <input type="number" class="contact-card__input mt-4 regular-font form-control" placeholder="شماره تماس">
+    <textarea class="contact-card__textarea form-control mt-4 regular-font" placeholder="پیام شما" rows="6"></textarea>
+    <button class="contact-card__submit py-2 mt-4 d-flex flex-row justify-content-center" data-bs-toggle="modal" data-bs-target="#staticBackdrop">
```
Comment (yeganemehr) [inline src/components/ContactCard.vue]: make it a submit button within the form (this make enter key and other OS features to work as expected)
don't forget to make spinner mode while waiting for server to response.

File: index.html (hunk 1 above)
Comment (behkha) [reply index.html]: installed bootstrap via npm

File: src/assets/main.css (hunk 2 above)
Comment (behkha) [reply src/assets/main.css:3]: used only IRANSandWeb(FaNum)

File: src/assets/main.css (hunk 3 above)
Comment (behkha) [reply src/assets/main.css]: used only IRANSandWeb(FaNum)

File: src/components/Brand.vue (hunk 4 above)
Comment (behkha) [reply src/components/Brand.vue:1]: Done

File: index.html (hunk 5 above)
Comment (behkha) [reply index.html:5]: favicon updated

File: src/components/DrawerMenu.vue (hunk 6 above)
Comment (behkha) [reply src/components/DrawerMenu.vue]: used href on MenuItem component

File: src/components/DrawerMenu.vue (hunk 7 above)
Comment (behkha) [reply src/components/DrawerMenu.vue]: used href on MenuItem component

File: src/components/DrawerMenu.vue (hunk 8 above)
Comment (behkha) [reply src/components/DrawerMenu.vue]: used href on MenuItem component

File: src/components/DrawerMenu.vue (hunk 9 above)
Comment (behkha) [reply src/components/DrawerMenu.vue]: used href on MenuItem component

File: src/components/DrawerMenu.vue (hunk 10 above)
Comment (behkha) [reply src/components/DrawerMenu.vue]: used href on MenuItem component

File: src/components/Footer.vue (hunk 11 above)
Comment (behkha) [reply src/components/Footer.vue]: Used Bootstrap tooltip

File: src/components/Footer.vue (hunk 12 above)
Comment (behkha) [reply src/components/Footer.vue]: Used Bootstrap tooltip

File: src/components/Footer.vue (hunk 13 above)
Comment (behkha) [reply src/components/Footer.vue]: Used Bootstrap tooltip

File: src/components/ContactCard.vue (hunk 14 above)
Comment (behkha) [reply src/components/ContactCard.vue]: Used type text on phone number input & validate value

File: src/components/ContactCard.vue (hunk 15 above)
Comment (behkha) [reply src/components/ContactCard.vue]: Added Validation

File: src/components/ContactCard.vue (hunk 16 above)
Comment (behkha) [reply src/components/ContactCard.vue]: Used @mdi/font

File: src/components/DrawerMenu.vue (hunk 17 above)
Comment (behkha) [reply src/components/DrawerMenu.vue]: Used @mdi/font

File: src/components/DrawerMenu.vue (hunk 18 above)
Comment (behkha) [reply src/components/DrawerMenu.vue]: Used @mdi/font

File: src/components/Menu.vue (hunk 19 above)
Comment (behkha) [reply src/components/Menu.vue]: Used @mdi/font

File: src/components/ContactCard.vue (hunk 20 above)
Comment (behkha) [reply src/components/ContactCard.vue]: Used submit button & added spinner on loading status.

File: src/assets/main.css (hunk 3 above)
Comment (yeganemehr) [reply src/assets/main.css]: As I describe it in telegram, use default fontiran example

File: src/App.vue [hunk 21]
```diff
@@ -1,25 +1,11 @@
...
-        <RouterLink to="/">Home</RouterLink>
-        <RouterLink to="/about">About</RouterLink>
-      </nav>
-    </div>
-  </header>
-
   <RouterView />
 </template>
 
+<script lang="ts">
```
Comment (yeganemehr) [inline src/App.vue]: This make an error:
https://github.com/jeodns/frontend/actions/runs/3919282572/jobs/6700194022

File: src/components/Brand.vue [hunk 22]
```diff
@@ -0,0 +1,29 @@
...
+});
+</script>
+
+<style lang="scss" scoped>
+a {
+  text-decoration: none;
+}
+
+p {
+  font-family: "Poppins";
```
Comment (yeganemehr) [inline src/components/Brand.vue]: What is this font? it's installed on all of OSes,like Arial?
It's better to change default font of your app to Iransans.

File: src/components/ContactCard.vue [hunk 23]
```diff
@@ -0,0 +1,239 @@
...
+      this.validateName()
+      this.validatePhoneNumber()
+      this.validateMessage()
+      if (this.isValid) {
+        this.$emit('submit', {
+          name: this.name,
+          phoneNumber: this.phoneNumber,
+          message: this.message
+        })
+        this.clearInputs()
```
Comment (yeganemehr) [inline src/components/ContactCard.vue]: if something goes wrong (for example server is down) all inputted data by user will cleared.
it's a very poor UX and can easily make user angry.

File: src/components/Content.vue [hunk 24]
```diff
@@ -0,0 +1,45 @@
...
+  &__container {
+    z-index: 3;
+  }
+
+  &__title {
+
+    font-size: 30px;
+    color: white;
+    text-align: right;
+    direction: rtl;
```
Comment (yeganemehr) [inline src/components/Content.vue]: why you code your entire app ltr and make some parts rtl?
make body rtl and text align right.

File: src/components/Footer.vue [hunk 25]
```diff
@@ -0,0 +1,57 @@
...
+</template>
+
+<script lang="ts">
+import { Tooltip } from "bootstrap";
+import { defineComponent } from "vue";
+
+export default defineComponent({
+  setup() {},
+  mounted() {
+    new Tooltip(this.$refs.telegram)
```
Comment (yeganemehr) [inline src/components/Footer.vue]: typing error in typescript.
test your code before you send PR.

File: src/components/Footer.vue [hunk 26]
```diff
@@ -0,0 +1,57 @@
+<template>
+  <div class="footer__container d-flex flex-row justify-content-end mx-5 mt-3">
+    <div class="footer-items__container d-flex flex-row my-auto mx-3">
+      <a class="footer__item d-flex flex-column align-items-center justify-items-center" ref="telegram" data-bs-toggle="tooltip" data-bs-placement="top" data-bs-title="Telegram" href="https://t.me/dnjco" target="_blank">
+        <img class="m-auto" src="@/assets/icons/telegram.svg" alt="">
```
Comment (yeganemehr) [inline src/components/Footer.vue]: I already asking you to change 4 images to font icons, should I ask you for every instance?
IF YOU CAN USE FONT ICONS, USE IT !!!
We living in iran, our customers and users living in Iran, network connection is a catastrophe, every http request took about 100-200 ms in best case scenario!

**Review by yeganemehr — CHANGES_REQUESTED** [review]

File: src/App.vue (hunk 21 above)
Comment (behkha) [reply src/App.vue]: Issue fixed

File: src/components/Brand.vue (hunk 22 above)
Comment (behkha) [reply src/components/Brand.vue]: Used IRANSans as default font

File: src/components/ContactCard.vue (hunk 23 above)
Comment (behkha) [reply src/components/ContactCard.vue]: Fixed: clearing inputs only on success

File: src/components/Content.vue (hunk 24 above)
Comment (behkha) [reply src/components/Content.vue]: Fixed: entire app is now rtl

File: src/components/Footer.vue (hunk 25 above)
Comment (behkha) [reply src/components/Footer.vue]: Fixed

File: src/components/Footer.vue (hunk 26 above)
Comment (behkha) [reply src/components/Footer.vue]: telegram icon was [removed](https://github.com/Templarian/MaterialDesign/issues/5275) from @mdi/fonts so i decided to use files instead to match exactly like UI.
But now i used mdi-send instead

**Review by yeganemehr — APPROVED** [review]

**Pull request merged**